From akpm@osdl.org Tue Oct 10 14:50:33 2006
Message-Id: <200610102142.k9ALgkt4025515@shell0.pdx.osdl.net>
Subject: USB: fix usbatm tiny race
From: Andrew Morton <akpm@osdl.org>
To: greg@kroah.com
Cc: akpm@osdl.org, baldrick@free.fr
Date: Tue, 10 Oct 2006 14:42:46 -0700

From: Andrew Morton <akpm@osdl.org>

ia64:

drivers/usb/atm/usbatm.c: In function `usbatm_do_heavy_init':
drivers/usb/atm/usbatm.c:1004: warning: implicit declaration of function `get_current'
drivers/usb/atm/usbatm.c:1004: error: invalid type argument of `->'

Signed-off-by: Duncan Sands <baldrick@free.fr>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/atm/usbatm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/atm/usbatm.c
+++ gregkh-2.6/drivers/usb/atm/usbatm.c
@@ -1001,7 +1001,7 @@ static int usbatm_do_heavy_init(void *ar
 
 	daemonize(instance->driver->driver_name);
 	allow_signal(SIGTERM);
-	instance->thread_pid = get_current()->pid;
+	instance->thread_pid = current->pid;
 
 	complete(&instance->thread_started);
 
